dipy/segment/clustering.py

```diff
@@ -0,0 +1,372 @@
+            elif op is operator.ne:
+                return not self == other
+
+            raise NotImplemented("Can only check if two ClusterMap instances are equal or not.")
+
+        elif isinstance(other, int):
+            return np.array([op(len(cluster), other) for cluster in self])
+
+        raise NotImplemented("ClusterMap only supports comparison with a int or another instance of Clustermap.")
+
+    def __eq__(self, other):
+        return self._richcmp(other, operator.eq)
+
+    def __ne__(self, other):
+        return self._richcmp(other, operator.ne)
+
+    def __lt__(self, other):
+        return self._richcmp(other, operator.lt)
+
+    def __le__(self, other):
+        return self._richcmp(other, operator.le)
+
+    def __gt__(self, other):
+        return self._richcmp(other, operator.gt)
+
+    def __ge__(self, other):
+        return self._richcmp(other, operator.ge)
+
+    def add_cluster(self, cluster):
+        """ Adds a new cluster to this cluster map.
+
+        Parameters
+        ----------
+        cluster : `Cluster` object
+            Cluster to be added in this cluster map.
+        """
+        self.clusters.append(cluster)
+        cluster.refdata = self.refdata
+
+    def remove_cluster(self, cluster):
+        """ Remove a cluster from this cluster map.
+
+        Parameters
+        ----------
+        cluster : `Cluster` object
+            Cluster to be removed from this cluster map.
+        """
+        self.clusters.remove(cluster)
+
+    def clear(self):
+        """ Remove all clusters from this cluster map. """
+        self.clusters.clear()
+
+
+class ClusterMapCentroid(ClusterMap):
+    """ Provides functionalities to interact with clustering outputs.
+
+    Useful container to create, remove, retrieve and filter clusters.
+    It allows also to retrieve easely the centroid of every clusters.
+    If `refdata` is given, elements will be returned instead of their
+    index when using `ClusterCentroid` objects.
+
+    Parameters
+    ----------
+    refdata : list
+        Actual elements that clustered indices refer to.
+    """
+    @property
+    def centroids(self):
+        return [cluster.centroid for cluster in self.clusters]
+
+
+class Clustering:
+    def cluster(self, data, ordering=None):
+        """ Clusters `data`.
+
+        Subclasses will perform their clustering algorithm here.
+
+        Parameters
+        ----------
+        data : list of N-dimensional array
+            Each array represents a data point.
+        ordering : iterable of indices, optional
+            Specifies the order in which data points will be clustered.
+
+        Returns
+        -------
+        `ClusterMap` object
+            Result of the clustering.
+        """
+        raise NotImplementedError("Subclass has to define method 'cluster(data, ordering)'!")
+
+
+class QuickBundles(Clustering):
+    r""" Clusters streamlines using QuickBundles algorithm.
```

Comment:
the QuickBundles algorithm or say using QuickBundles. 

And link to citation is missing.